configurate-core/src/main/java/ninja/leaping/configurate/objectmapping/ObjectMapper.java

```diff
@@ -75,23 +76,36 @@
      * Holder for field-specific information
      */
     protected static class FieldData {
-        private final Field field;
-        private final TypeToken<?> fieldType;
-        private final String comment;
+        final Field field;
+        final TypeToken<?> fieldType;
+        @Nullable final TypeSerializer<?> customSerializer;
+        @Nullable final String comment;
 
-        public FieldData(Field field, String comment) throws ObjectMappingException {
+        @Deprecated
+        public FieldData(Field field, @Nullable  String comment) {
+            this(field, comment, null);
+        }
+
+        public FieldData(Field field, @Nullable  String comment, @Nullable TypeSerializer<?> customSerializer) {
             this.field = field;
             this.comment = comment;
             this.fieldType = TypeToken.of(field.getGenericType());
+            this.customSerializer = customSerializer;
         }
 
         public void deserializeFrom(Object instance, ConfigurationNode node) throws ObjectMappingException {
-            TypeSerializer<?> serial = node.getOptions().getSerializers().get(this.fieldType);
-            if (serial == null) {
-                throw new ObjectMappingException("No TypeSerializer found for field " + field.getName() + " of type "
-                        + this.fieldType);
+            Object newVal = null;
```

Comment:
No need to separate out the declaration and assignment of the `newVal` variable. This can be moved back down to line 108.